### PR TITLE
Add ability to filter find on mode

### PR DESCRIPTION
**Pull request by sivel**
##### SUMMARY

Add ability to filter find on mode

##### ISSUE TYPE

- Feature Pull Request

##### ADDITIONAL INFORMATION

AAPRFE-32

<details>
<summary>Examples...</summary>
<p>

```yaml
    - name: find executables with octal
      find:
        path: '{{ playbook_dir }}'
        mode: '0o111'
        exact_mode: no

    - name: find executable with symbolic
      find:
        path: '{{ playbook_dir }}'
        mode: 'a+x'
        exact_mode: no

    - name: find readable files with octal
      find:
        path: '{{ playbook_dir }}'
        mode: '0o444'
        exact_mode: no

    - name: find readable files with symbolic
      find:
        path: '{{ playbook_dir }}'
        mode: 'a+r'
        exact_mode: no

    - name: find files with at least 644 octal perms
      find:
        path: '{{ playbook_dir }}'
        mode: '0o644'
        exact_mode: no

    - name: find files with exactly 644 octal perms
      find:
        path: '{{ playbook_dir }}'
        mode: '0o644'
        exact_mode: yes
```

</p>
</details>

### CODE REVIEW

File: lib/ansible/modules/find.py [hunk 1]
```diff
@@ -344,6 +366,26 @@ def contentfilter(fsname, pattern, read_whole_file=False):
     return False
 
 
+def mode_filter(st, mode, exact, module):
+    if not mode:
+        return True
+
+    st_mode = stat.S_IMODE(st.st_mode)
+
+    if not isinstance(mode, int):
```
Comment (bcoca) [inline lib/ansible/modules/find.py]: I still think we should not accept ints, using raw you can still implement 'strict strings' check on module side

Comment (sivel) [reply lib/ansible/modules/find.py]: Done.

**Review by bcoca — APPROVED** [review]
thinking i need to redesign this to a filter queue that gets compared at each step vs having to add the same filters x10 in the code.

not blocking this pr, just note for future bcoca when he complains that there should be better way.

File: lib/ansible/modules/find.py [hunk 2]
```diff
@@ -111,6 +111,22 @@
             - Set this to V(true) to include hidden files, otherwise they will be ignored.
         type: bool
         default: no
+    mode:
+        description:
+            - Choose objects matching a specified permission. This value is
+              restricted to modes that can be applied using the python
+              C(os.chmod) function.
+            - The mode can be provided as an octal such as V("0o0644") or
```
Comment (sivel) [inline lib/ansible/modules/find.py]:
```suggestion
            - The mode can be provided as an octal such as V("0644") or
```

File: lib/ansible/modules/find.py [hunk 3]
```diff
@@ -414,12 +456,19 @@ def main():
...
             depth=dict(type='int'),
+            mode=dict(type='raw'),
+            exact_mode=dict(type='bool', default=True),
         ),
         supports_check_mode=True,
     )
 
     params = module.params
 
+    if params['mode'] and not isinstance(params['mode'], string_types):
```
Comment (whoami-anoint) [inline lib/ansible/modules/find.py:467]:
```
if 'mode' in params and not isinstance(params['mode'], str):
    module.fail_json(
        msg="Argument 'mode' is not a string and conversion is not allowed. Value is of type %s." % type(params['mode']).__name__
    )
```

**Review by whoami-anoint — COMMENTED** [review]
Used the in keyword to explicitly check if 'mode' is present in the params dictionary. This helps prevent a KeyError if 'mode' is not present in the dictionary.

**Pull request merged**